sdks/java/core/src/main/java/org/apache/beam/sdk/io/fs/MoveOptions.java

```diff
@@ -17,16 +17,61 @@
  */
 package org.apache.beam.sdk.io.fs;
 
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
 import org.apache.beam.sdk.io.FileSystems;
 
 /**
  * An object that configures {@link FileSystems#copy}, {@link FileSystems#rename}, and {@link
  * FileSystems#delete}.
  */
-public interface MoveOptions {
+public abstract class MoveOptions {
+
+  /**
+   * If false, operations will abort on errors arising from missing files. Otherwise, operations
+   * will ignore such errors.
+   */
+  public abstract boolean getIgnoreMissingFiles();
+
+  /**
+   * Specifies the Key Management System key name to use to encrypt new files with.
+   *
+   * <p>Only relevant to operations that create new files and filesystems that support KMS features.
+   */
+  @Experimental(Kind.FILESYSTEM)
+  @Nullable
+  public abstract String getDestKmsKey();
```

Comment:
This should not be done here but better in the FileSystem options. Notice also that you may need to set up a different SourceKmsKey so this should be considered.